analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs

```diff
@@ -31,6 +31,8 @@ public abstract class MutableFieldsShouldNotBe : SonarDiagnosticAnalyzer
             ImmutableArray.Create(
                 KnownType.System_Collections_ObjectModel_ReadOnlyCollection_T,
                 KnownType.System_Collections_ObjectModel_ReadOnlyDictionary_TKey_TValue,
+                KnownType.System_Collections_Frozen_FrozenDictionary_TKey_TValue,
+                KnownType.System_Collections_Frozen_FrozenSet_T,
                 KnownType.System_Collections_Immutable_IImmutableArray_T,
```

Comment:
I suspect that the problem is that we are missing `IReadOnly` interfaces in this list.
If we had those, wouldn't frozen collections work without the need to be explicitly mentioned in the list?
However, I don't understand, just by looking at the code, how the following cases work:

```cs
public class MyReadOnlySet : IReadOnlySet<string>
{
    public bool Contains(string item) => throw new NotImplementedException();
    public int Count => throw new NotImplementedException();
    public IEnumerator<string> GetEnumerator() => throw new NotImplementedException();
    System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() => throw new NotImplementedException();
    public bool IsProperSubsetOf(IEnumerable<string> other) => throw new NotImplementedException();
    public bool IsProperSupersetOf(IEnumerable<string> other) => throw new NotImplementedException();
    public bool IsSubsetOf(IEnumerable<string> other) => throw new NotImplementedException();
    public bool IsSupersetOf(IEnumerable<string> other) => throw new NotImplementedException();
    public bool Overlaps(IEnumerable<string> other) => throw new NotImplementedException();
    public bool SetEquals(IEnumerable<string> other) => throw new NotImplementedException();
}

public class MyReadOnlyList: IReadOnlyList<string>
{
    public string this[int index] => throw new NotImplementedException();
    public int Count => throw new NotImplementedException();
    public IEnumerator<string> GetEnumerator() => throw new NotImplementedException();
    System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() => throw new NotImplementedException();
}

public class MyList: List<string>
{
}

public class FrozenCollections
{
    public static MyReadOnlyList MyReadonlyListTest = new MyReadOnlyList(); // Compliant => Correct but how?
    public static MyReadOnlySet MyReadonlySetTest = new MyReadOnlySet(); // Compliant => Correct but how?
    public static MyList MyListTest = new MyList(); // Noncompliant => OK
}
```

Reply:
> I suspect that the problem is that we are missing IReadOnly interfaces in this list.

See the comment above: that interface is implemented by mutable and immutable collections alike.

Reply:
Got it, thanks!